Prefect3 pixi upgrade

JunAishima opened the pull request.

JunAishima committed:
initial commit of files for Prefect3/Pixi/Podman update

JunAishima committed:
fix pyxrf to same version as previous conda env

 * collection conda env was 2024-2.3-py310-tiled
 * update pixi.lock file to reflect this change

JunAishima committed:
replace tst with srx

 * add version number
 * remove access token requirement for code repo
 * point to nsls-ii-srx workflows repo

JunAishima committed:
for initial testing, use current branch

JunAishima committed:
add databroker

JunAishima committed:
add /repo, fix userns settings, update version

JunAishima committed:
update location to pull from

 * currently the workflow repo in my account

JunAishima committed:
add missing file

JunAishima committed:
remove adding overlay directory to PATH

 * replaced by adding ot pixi dependencies

JunAishima committed:
update pixi lock file

JunAishima committed:
update tiled to more recent version

 * fix import error:
>>> from tiled.client.composite import CompositeClient
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: cannot import name 'CompositeClient' from 'tiled.client.composite' (/root/code/srx-workflows/.pixi/envs/default/lib/python3.12/site-packages/tiled/client/composite.py)

JunAishima committed:
fix syntax error

JunAishima committed:
update Prefect where Slackwebhook issue is fixed

 * https://github.com/PrefectHQ/prefect/issues/8517

JunAishima committed:
updated lock file after prefect version update

JunAishima committed:
fix time zone name

 * now getting complaints about time zone name, so fix this

JunAishima committed:
fix deployment name by updating the wrapper function name

 * the flow name is automatically scraped from the function that the
   @flow decorator is attached to, but this leads to using the wrapper
   function name in the case of a decorator like @slack
 * for now, change the name of the wrapper to end_of_run_workflow to
   match our expectations

JunAishima committed:
update code location to NSLS-II-SRX organization

JunAishima committed:
update pyxrf to use srx metadata update

JunAishima committed:
add file for building and publishing container image with actions

JunAishima committed:
add missing dependency

 * pyxrf complains about this, though it is in not in the list of
   requirements. need to get clarification on this. this has changed
   from the previous installed version, which was 1.0.31

JunAishima committed:
update pixi.lock

JunAishima committed:
hard-code repo name to match prefect deployment name

JunAishima committed:
add org name to image name

JunAishima committed:
add scikit-beam dependency

 * also required for pyxrf, but not in requirements.txt - is in the
   pyxrf feedstock recipe

JunAishima committed:
remove build/push from prefect.yaml file

 * use the github action to build/publish image instead

AbbyGi commented on prefect.yaml:
```diff
@@ -1,46 +1,56 @@
-  schedule:
-  is_schedule_active: true
+  - name: srx-end-of-run-workflow-docker
+    version: 0.1.1
```
Additional tags for beamline/endstation and branch to keep things consistent.
```suggestion
    version: 0.1.1
    tags:
      - srx
      - main
```

AbbyGi commented on end_of_run_workflow.py:
```diff
@@ -20,7 +20,7 @@ def slack(func):
     """
 
-    def wrapper(stop_doc):
+    def end_of_run_workflow(stop_doc):
```
Calling this inner function `end_of_run_workflow` could get confusing with the actual end of run workflow that the beamline scientists should be updating (if they want to update their own code). Maybe this inner function should stay as `wrapper` for now and we can move the `slack` function to it's own utility file.

JunAishima replied:
This comes from https://github.com/JunAishima/srx-workflows/commit/82c751796241d949dc307e6258a447328624dff9 where I found that Prefect was getting the flow name from the function, regardless of whether it was the wrapper for another function or not. I will update this function to clarify the reason for this naming.

JunAishima replied:
added explanation in https://github.com/NSLS2/srx-workflows/pull/28/commits/27078038fb7530d386f54e0bcce3e02ad2ec37d4

AbbyGi reviewed (commented):
General HEX comments + two additional comments

JunAishima committed:
update branch name and repo name

JunAishima committed:
Update prefect.yaml

Co-authored-by: Abby Giles <35899293+AbbyGi@users.noreply.github.com>

JunAishima committed:
remove run-post-link-script option

 * no problems observed when running workflows when this option
   is removed. only librsvg is affected but does not appear to be used
   when running a workflow

JunAishima committed:
remove containers after running

 * prevent worker host from filling with old containers

JunAishima committed:
add explanation for the seemingly duplicate naming

 * Prefect, when introspecting for the flow name, was getting the
   wrapper function's name instead of end_of_run_workflow

JunAishima committed:
go back to prefect3-pixi-upgrade branch

 * start generating docker images now

JunAishima committed:
remove unnecessary text

 * add extra line to match other files

JunAishima committed:
remove ENV field from Dockerfile

JunAishima committed:
do not put Tiled API key in deployment

JunAishima committed:
refactor to use common get_tiled_client() function

JunAishima committed:
Merge remote-tracking branch 'jun/prefect3-pixi-upgrade' into prefect3-pixi-upgrade

AbbyGi reviewed (approved):
Looks good to me.

JunAishima committed:
add fix to problem during deployment

 * got a coroutine error when attempting to do 'prefect deploy'. adding
   this kwarg to the load function was proposed by Prefect to fix this

JunAishima committed:
ensure latest image is pulled

JunAishima committed:
Revert "refactor to use common get_tiled_client() function"

This reverts commit c0704dd854022344d1d4b4aca96ff88a5304fc1e.

JunAishima committed:
get key from Prefect Secret for Tiled client

JunAishima committed:
get key from Prefect Secret for Tiled client

JunAishima committed:
add new deployment file for Prefect2

JunAishima committed:
add flag to enable clean deployment

JunAishima committed:
ensure using Python < 3.13

 * Prefect is not guaranteed to work with this version yet

JunAishima committed:
add Tiled API key as an env var for pyxrf

JunAishima committed:
point to main

 * after merging, the building and pulls are from the main branch
